dataRetriever/resolvers/miniblockResolver.go

```diff
@@ -0,0 +1,202 @@
+	Marshalizer      marshal.Marshalizer
+	AntifloodHandler dataRetriever.P2PAntifloodHandler
+	Throttler        dataRetriever.ResolverThrottler
+	DataPacker       dataRetriever.DataPacker
+}
+
+// miniblockResolver is a wrapper over Resolver that is specialized in resolving miniblocks requests
+type miniblockResolver struct {
+	dataRetriever.TopicResolverSender
+	messageProcessor
+	miniBlockPool    storage.Cacher
+	miniBlockStorage storage.Storer
+	dataPacker       dataRetriever.DataPacker
+}
+
+// NewMiniblockResolver creates a miniblock resolver
+func NewMiniblockResolver(arg ArgMiniblockResolver) (*miniblockResolver, error) {
+	if check.IfNil(arg.SenderResolver) {
+		return nil, dataRetriever.ErrNilResolverSender
+	}
+	if check.IfNil(arg.MiniBlockPool) {
+		return nil, dataRetriever.ErrNilMiniblocksPool
+	}
+	if check.IfNil(arg.MiniBlockStorage) {
+		return nil, dataRetriever.ErrNilMiniblocksStorage
+	}
+	if check.IfNil(arg.Marshalizer) {
+		return nil, dataRetriever.ErrNilMarshalizer
+	}
+	if check.IfNil(arg.AntifloodHandler) {
+		return nil, dataRetriever.ErrNilAntifloodHandler
+	}
+	if check.IfNil(arg.Throttler) {
+		return nil, dataRetriever.ErrNilThrottler
+	}
+	if check.IfNil(arg.DataPacker) {
+		return nil, dataRetriever.ErrNilDataPacker
+	}
+
+	mbResolver := &miniblockResolver{
+		TopicResolverSender: arg.SenderResolver,
+		miniBlockPool:       arg.MiniBlockPool,
+		miniBlockStorage:    arg.MiniBlockStorage,
+		dataPacker:          arg.DataPacker,
+		messageProcessor: messageProcessor{
+			marshalizer:      arg.Marshalizer,
+			antifloodHandler: arg.AntifloodHandler,
+			topic:            arg.SenderResolver.RequestTopic(),
+			throttler:        arg.Throttler,
+		},
+	}
+
+	return mbResolver, nil
+}
+
+// ProcessReceivedMessage will be the callback func from the p2p.Messenger and will be called each time a new message was received
+// (for the topic this validator was registered to, usually a request topic)
+func (mbRes *miniblockResolver) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPeer p2p.PeerID) error {
+	err := mbRes.canProcessMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	mbRes.throttler.StartProcessing()
+	defer mbRes.throttler.EndProcessing()
+
+	rd, err := mbRes.parseReceivedMessage(message)
+	if err != nil {
+		return err
+	}
+
+	switch rd.Type {
+	case dataRetriever.HashType:
+		var buff []byte
+		buff, err = mbRes.resolveMbRequestByHash(rd.Value)
+		if err != nil {
+			return err
+		}
+		return mbRes.Send(buff, message.Peer())
+	case dataRetriever.HashArrayType:
+		return mbRes.resolveMbRequestByHashArray(rd.Value, message.Peer())
+	default:
+		return dataRetriever.ErrRequestTypeNotImplemented
+	}
+}
+
+func (mbRes *miniblockResolver) resolveMbRequestByHash(hash []byte) ([]byte, error) {
+	mb, err := mbRes.fetchMbAsByteSlice(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	buffToSend, err := mbRes.marshalizer.Marshal(&batch.Batch{Data: [][]byte{mb}})
```

Comment:
refactor that Batch initialization to multiple lines, maybe even a new variable

Reply:
done

Reply:
not done yet